dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/DefaultAnalyticsTableService.java

```diff
@@ -122,60 +124,88 @@ public void create(AnalyticsTableUpdateParams params, JobProgress progress) {
         toLongDate(params.getFromDate()),
         params);
     progress.startingStage("Performing pre-create table work");
-    progress.runStage(() -> tableManager.preCreateTables(params));
-    clock.logTime("Performed pre-create table work " + tableType);
 
+    //
+    // Perform pre-create tables
+    //
+    withClock(
+        clock,
+        "Performed pre-create table work " + tableType,
+        () -> analyticsTableStrategy.preCreateTables(params, progress));
+
+    //
+    // Drop staging tables
+    //
     progress.startingStage("Dropping staging tables (if any) " + tableType, tables.size());
-    dropTables(tables, progress);
-    clock.logTime("Dropped staging tables");
-
+    withClock(
+        clock,
+        "Dropped staging tables " + tableType,
+        () -> analyticsTableStrategy.dropStagingTables(tables, progress));
+
+    //
+    // Create analytics tables
+    //
     progress.startingStage("Creating analytics tables " + tableType, tables.size());
-    createTables(tables, progress);
-    clock.logTime("Created analytics tables");
+    withClock(
```

Comment:
I am wondering if `withClock` makes the code a bit harder to read. Adding timing info is sort of a secondary activity.

Reply:
Reverted